src/core/Akka/Actor/Futures.cs

```diff
@@ -322,6 +326,46 @@ public override int GetHashCode()
         // use a static delegate to avoid allocations
         private static readonly Action<object> CancelAction = o => ((TaskCompletionSource<object>)o).TrySetCanceled();
 
+        /// <summary>
+        /// Creates a new <see cref="PromiseActorRef"/>
+        /// </summary>
+        /// <param name="provider">The current actor ref provider.</param>
+        /// <param name="messageClassName">The name of the message class.</param>
+        /// <param name="cancellationToken">An external cancellation token.</param>
+        /// <returns>A new <see cref="PromiseActorRef"/></returns>
+        /// <remarks>
+        /// API is used by WatchAsync.
+        /// </remarks>
+        public static PromiseActorRef Apply(IActorRefProvider provider,
```

Comment:
Primary method now called by both `GracefulStop` and `WatchAsync` - I really didn't want to have users pass in a `TimeSpan` on `WatchAsync` because those tasks are _supposed_ to be long-lived, therefore a `CancellationToken` was more appropriate. So I added a new overload and synced the previously used method to call it internally.